test/e2e_node/runtime_conformance_test.go

```diff
@@ -28,111 +32,319 @@ import (
+
+			BeforeEach(func() {
+				testCContainers = []ConformanceContainer{}
+			})
+
+			It("it should run with the expected status [Conformance]", func() {
+				testContainer := api.Container{
+					Image:           "gcr.io/google_containers/busybox",
+					Command:         []string{"sh", "-c", "while true; do echo hello; sleep 1; done"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				}
+				testStatuss := []testStatus{
+					{"loop-always", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, "==", 0, true},
+					{"loop-onfailure", api.RestartPolicyOnFailure, api.PodRunning, ContainerStateRunning, "==", 0, true},
+					{"loop-never", api.RestartPolicyNever, api.PodRunning, ContainerStateRunning, "==", 0, true},
+				}
+				for _, testStatus := range testStatuss {
+					testContainer.Name = testStatus.Name
+					runningContainer := ConformanceContainer{
+						Container:     testContainer,
+						Client:        cl,
+						RestartPolicy: testStatus.RestartPolicy,
+						NodeName:      *nodeName,
+						Namespace:     namespace,
+					}
+					err := runningContainer.Create()
+					Expect(err).NotTo(HaveOccurred())
+					testCContainers = append(testCContainers, runningContainer)
+
+					Eventually(func() api.PodPhase {
+						_, phase, _ := runningContainer.GetStatus()
+						return phase
+					}, retryTimeout, pollInterval).Should(Equal(api.PodRunning))
+
+					var status api.ContainerStatus
+					var phase api.PodPhase
+					By("it should get the expected 'Phase'")
+					Consistently(func() api.PodPhase {
+						status, phase, err = runningContainer.GetStatus()
+						return phase
+					}, retryTimeout, pollInterval).Should(Equal(testStatus.Phase))
+					Expect(err).NotTo(HaveOccurred())
+
+					By("it should get the expected 'RestartCount'")
+					Expect(status.RestartCount).To(BeNumerically(testStatus.RestartCountOper, testStatus.RestartCount))
+
+					By("it should get the expected 'Ready' status")
+					Expect(status.Ready).To(Equal(testStatus.Ready))
+
+					By("it should get the expected 'State'")
+					Expect(GetContainerState(status.State) & testStatus.State).NotTo(Equal(0))
+
+					By("it should be possible to delete [Conformance]")
+					err = runningContainer.Delete()
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(func() bool {
+						isPresent, err := runningContainer.Present()
+						return err == nil && !isPresent
+					}, retryTimeout, pollInterval).Should(BeTrue())
+				}
 			})
 
 			AfterEach(func() {
-				containerCase.Delete()
-				Eventually(func() bool {
-					isPresent, err := containerCase.Present()
-					return err == nil && !isPresent
-				}, retryTimeout, pollInterval).Should(BeTrue())
+				for _, cc := range testCContainers {
+					cc.Delete()
+				}
+			})
+		})
+
+		Context("when start a container that exits failure", func() {
+			var testCContainers []ConformanceContainer
+
+			BeforeEach(func() {
```

Comment:
Why not share the `BeforeEach` and `AfterEach` between these contexts. They are exactly the same.